core/src/sumeragi/mod.rs

```diff
@@ -857,17 +859,20 @@ impl<G: GenesisNetworkTrait, K: KuraTrait, W: WorldTrait, F: FaultInjection>
             info!(?event);
             drop(self.events_sender.send(event));
         }
+        let signed_block = block.sign(self.key_pair.clone())?;
         if !network_topology.is_consensus_required() {
-            self.commit_block(block).await;
```

Comment:
this was a bug previously because block wasn't signed by the leader

Reply:
Good catch.

Reply:
test started failing all of a sudden when I enforced `SignaturesOf` invariant to contain at least one signature